src/OpenSkos/Institution/Controller/InstitutionController.php

```diff
@@ -105,22 +148,99 @@ public function postInstitution(
             $errors = array_merge($errors, $institution->errors());
         }
         if (count($errors)) {
-            throw new \Exception(json_encode($errors));
+            foreach ($errors as $error) {
+                throw new ApiException($error);
+            }
         }
 
-        // Check if the resources already exist
+        // Save all given institutions
         foreach ($institutions as $institution) {
-            if ($institution->exists()) {
-                throw new ConflictHttpException('Resource [id] already exists');
+            $institution->save();
+        }
+
+        // Return re-fetched institutions
+        return new ListResponse(
+            $institutions,
+            count($institutions),
+            0,
+            $apiRequest->getFormat()
+        );
+    }
+
+    /**
+     * @Route(path="/institution/{id}.{format?}", methods={"DELETE"})
+     */
+    public function deleteInstitution(
+        InternalResourceId $id,
+        ApiRequest $apiRequest,
+        InstitutionRepository $repository
+    ): ScalarResponse {
+        // Client permissions
+        $auth = $apiRequest->getAuthentication();
+        $auth->requireAdministrator();
+
+        // Fetch the institution we're deleting
+        /** @var AbstractRdfDocument $institution */
+        $institution = $this->getInstitution($id, $apiRequest, $repository)->doc();
+
+        $institution->delete();
+
+        return new ScalarResponse(
+            $institution,
+            $apiRequest->getFormat()
+        );
+    }
+
+    /**
+     * @Route(path="/institutions.{format?}", methods={"PUT"})
+     *
+     * @Error(code="institution-update-empty-or-corrupt-body",
+     *        status=400,
+     *        description="The body passed to this endpoint was either missing or corrupt"
+     * )
+     * @Error(code="institution-update-does-not-exist",
+     *        status=400,
+     *        description="The institution with the given iri does not exist",
+     *        fields={"iri"}
+     * )
+     */
+    public function updateInstitution(
```

Comment:
Should become putInstitution, for consistency with the other controller functions.